tools/2024-09-gitops-flux.md

```diff
@@ -0,0 +1,289 @@
+will be follow-ups after this work, list them here. If there are things
+we plan to do in the future, but are out of scope of this design, list
+them here. Provide a brief explanation on why this is a non-goal.
+-->
+
+**Non-goal: Support for other GitOps tools**
+This design is focused on supporting Flux. Support for other GitOps tools (such as ArgoCD) is out of scope for this design and will be considered in a future design. The design principles outlined here may be applicable to other GitOps tools, but the implementation details will be specific to Flux.
+
+### User scenarios
+
+<!--
+Describe the user scenarios for this design. Ensure that you define the
+roles and personas in these user scenarios when it requires API design.
+If you have an existing issue that describes the user scenarios, please
+link to that issue instead.
+-->
+
+#### Jon can use GitOps to deploy a standard Radius environment for developer use
+
+Jon is an infrastructure operator who uses GitOps to automate the creation of developer Kubernetes clusters. Today, he publishes Kubernetes manifests to a GitHub repository, and a GitOps tool watches the repository and applies the manifests to the Kubernetes cluster. He wants to have a Radius environment and a set of recipes available on the clusters for his developers to use, since they are building applications using Radius.
+
+He first creates a GitHub repository that contains the cluster configuration and the Bicep manifests for the Radius environment. He includes the installation of Radius and the registration of Radius environment and recipes for the developers to use. He then configures the GitOps tool to watch the repository and apply the manifests to the Kubernetes cluster.
+
+When a developer creates a new development cluster using the configuration specified in Jon's GitHub repository with a GitOps tool, Radius is automatically installed and the environment and recipes are pre-configured. The developer can then use the environment to deploy and test their applications using Radius.
+
+#### Bran can use GitOps to deploy and manage a Radius application in production
+
+Bran is a developer who uses GitOps to automate the deployment of Radius applications to production Kubernetes clusters. His infrastructure team has set up a Radius installation and environment on the production cluster, and Bran wants to deploy his application using Radius. He has authored a Bicep manifest that defines the resources for his application and a Bicep parameters file that specifies the parameters for the manifest. He commits the Bicep manifest and parameters file to a Git repository.
+
+When the GitOps tool watches the repository and applies the Bicep manifest and parameters file to the production cluster, the Radius resources defined in the Bicep manifest are deployed. The application is deployed and Bran can manage the application using GitOps practices.
+
+#### Sam can use GitOps to patch a Radius application in production
```

Comment:
A better user story would be that Sam receives an alert warning that application latency is above the error budget. In order to improve latency, Sam needs to modify an auto scaling parameter to increase the maximum number of replicas.

But it looks like Radius only supports [manual scaling](https://docs.radapp.io/reference/resource-schema/core-schema/container-schema/#manualscaling), so maybe your example is fine for now. I need to look at how we support horizontal pod autoscaling.